LanguageExt.Core/Stack.cs

```diff
@@ -840,6 +850,16 @@ public static class StackExtensions
         LanguageExt.List.distinct(stack, compare);
 
     /// <summary>
+    /// Return an enumerable with all duplicate values removed
+    /// </summary>
+    /// <typeparam name="T">Stack item type</typeparam>
+    /// <param name="stack">Stack</param>
+    /// <returns>An enumerable with all duplicate values removed</returns>
+    [Pure]
+    public static IEnumerable<T> Distinct<T, K>(Stck<T> stack, Func<T, K> keySelector, Option<Func<K, K, bool>> compare = default(Option<Func<K, K, bool>>)) =>
+        LanguageExt.List.distinct(stack, keySelector, compare);
```

Comment:
Is this needed?  `Stck<T>` is already an `IEnumerable`, and therefore the `Distinct` extension method for `IEnumerable` will do the same job.  If it is needed, then it should be `this Stck<T>` as the first argument.